### ISSUE TITLE
[Excel] How can I update the text format only for the highlighted parts?

### ISSUE THREAD

**Issue by Eclipse-Walker** [OfficeDev/office-js#3802]
## How can I update the text format only for the highlighted parts?

I want to update the text format within a cell and update only the highlighted text.
![Screenshot_1](https://github.com/OfficeDev/office-js/assets/42166625/0d3b5e04-60c9-4ffe-9ac1-992f3be4ea36)

Now I don't know how to update only the highlighted messages. I have a sample source The code I'm using is updating the entire cell, which is not what I want.
```
Excel.run(function(context) {
  var sheet = context.workbook.worksheets.getActiveWorksheet();
  var range = sheet.getRange("A1"); 

  return context.sync()
    .then(function() {      
      range.format.font.superscript = true;
    })
    .then(context.sync);
}).catch(function(error) {
  console.log(error);
});
```
The result from the code above
![Screenshot_3](https://github.com/OfficeDev/office-js/assets/42166625/92972ad4-2961-4628-891a-a0383fbcbc01)

What I expect is to update only the highlighted messages, and below is my expected result.
![Screenshot_2](https://github.com/OfficeDev/office-js/assets/42166625/ab85a179-78ca-4ac9-a784-e47ab19f24fd)

Please give me some advice. Thank you in advance.

**Comment by dingjin-ms** [comment]
Hi @I3ushido , we apologize for the inconvenience, but we do not support the scenario you mentioned at the moment. If you would like to request this feature, please post it on our [developer forum](https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365DeveloperPlatform), where our product managers will regularly review and prioritize the suggestions. Thank you for your feedback and interest in our product!

**Comment by Eclipse-Walker** [comment]
Thank you for your information I do appreciate your prompt reply.

**Comment by adrianwu8516** [comment]
Hi @Eclipse-Walker , 

Thank you very much for the suggestion. As @dingjin-ms mentioned, we currently do not support such operations at the API level. If you are expecting such functionality, it would be very helpful if you could help us create a new post in [our technical forum](https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365DeveloperPlatform) so that more people with similar requests can join the discussion and upvote for this idea. 

Given the circumstances, I will be closing this issue to keep our tracker focused on immediate actionable items. We appreciate your engagement and look forward to your contributions on the forum!

Thank you again for your suggestion and your active participation.
 
> Microsoft 365 Developer Platform: 
https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365DeveloperPlatform

**Comment by adrianwu8516** [comment]
Hi @Eclipse-Walker,

Thank you for your patience. The requested rich text support feature has been implemented and will be included in the upcoming Excel requirement set 1.18 release, scheduled for February 2025. In the meantime, you can already experiment with these capabilities in the beta build.

The following APIs will be available:

- `Excel.CellPropertiesInternal.TextRuns`
- `Excel.CellPropertiesLoadOptionsInternal.TextRuns`
- `Excel.RangeTextRun.Text`
- `Excel.RangeTextRun.Font`

For more details, please refer to the [official documentation](https://learn.microsoft.com/en-us/javascript/api/excel/excel.rangetextrun?view=excel-js-preview). Below is a sample code snippet that demonstrates how you can use rich text formatting in cells:

```javascript
(async () => {
  await Excel.run(async (context) => {
    var sheet = context.workbook.worksheets.getItem("TestSheet");
    const cellSrc = sheet.getRange("A1");
    const cellSrcTextRun: Excel.SettableCellProperties = {
      textRuns: [
        { text: "Sample", font: { bold: true, color: "#00B0F0", size: 14, italic: true } },
        { text: "1", font: { subscript: true } },
        { text: "String" },
        { text: "2", font: { superscript: true, color: "black", tintAndShade: 0.5 } },
      ],
    };

    cellSrc.clear(Excel.ClearApplyTo.all);
    cellSrc.setCellProperties([[cellSrcTextRun]]);
    await context.sync();

    const textRunCellProperty = cellSrc.getCellProperties({ textRuns: true });
    await context.sync();

    const cellTextRuns = textRunCellProperty.value[0][0].textRuns;
    console.log(JSON.stringify(cellTextRuns, undefined, "  "));
  });
})();
```

If you have any further questions or feedback, feel free to let us know!
